src/azure-cli/azure/cli/command_modules/storage/operations/azcopy.py

```diff
@@ -5,13 +5,89 @@
 
 from __future__ import print_function
 from ..azcopy.util import AzCopy, client_auth_for_azcopy, login_auth_for_azcopy
-
-
-def storage_blob_copy(azcopy, source, destination, recursive=None):
+from azure.cli.command_modules.storage._client_factory import blob_data_service_factory, file_data_service_factory
+
+
+def storage_copy(cmd, source=None,
+                 destination=None,
+                 put_md5=None,
+                 recursive=None,
+                 source_account_name=None,
+                 source_container=None,
+                 source_blob=None,
+                 source_share=None,
+                 source_file_path=None,
+                 source_local_path=None,
+                 destination_account_name=None,
+                 destination_container=None,
+                 destination_blob=None,
+                 destination_share=None,
+                 destination_file_path=None,
+                 destination_local_path=None):
+    def get_url_with_sas(source, account_name, container, blob, share, file_path, local_path):
+        import re
+        import os
+        from azure.cli.command_modules.storage._validators import _query_account_key
+        from azure.cli.command_modules.storage.azcopy.util import _generate_sas_token
+
+        if source is not None:
+            if "?" in source:  # sas token exists
+                return source
+            storage_pattern = re.compile(r'https://(.*?)\.(blob|dfs|file).core.windows.net')
```

Comment:
The storage endpoints are different for different cloud. This pattern here will only work for public cloud. You need to support other clouds. See other clouds' endpoints here: https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/cloud.py#L154